sdk/core/System.ClientModel/samples/Diagnostics.md

```diff
@@ -1,6 +1,81 @@
-# System.ClientModel-based client logging samples
+# Diagnostics in System.ClientModel-based clients
+
+Instrumentation is an essential part of developing client libraries. System.ClientModel leverages existing .NET observability APIs and tools to provide building blocks that are tailored to client-specific needs.
+
+## Distributed tracing
+
+### Adding distributed tracing instrumentation to service clients
+
+System.ClientModel provides APIs for library authors to use to create distributed tracing spans in each public method call. These APIs are extensions on System.Diagnostics.Activity and System.Diagnostics.ActivitySource. See [Add distributed tracing instrumentation](https://learn.microsoft.com/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs) for more information about these APIs and how to further customize telemetry using them.
+
+The following sample shows an example of instrumenting a service client implementation.
+
+```C# Snippet:OpenTelemetryInClient
+public class SampleClient
+{
+    private readonly Uri _endpoint;
+    private readonly ApiKeyCredential _credential;
+    private readonly ClientPipeline _pipeline;
+    private readonly SampleClientOptions _sampleClientOptions;
+
+    // Each client should have a static ActivitySource named after the full name
+    // of the client.
+    // Most of the time, tracing should start as experimental.
```

Comment:
Just curious - What is the value of adding the `Experimental` prefix here?

Reply:
Azure clients have stable conventions - https://github.com/Azure/azure-sdk/blob/main/docs/tracing/distributed-tracing-conventions.md#public-api-calls 
For non-Azure the client libraries most likely wouldn't have stable conventions so in that case it would be experimental. The prefix forces listeners to opt into it being experimental when the listener is started and the ActivitySource name is passed to the listener.

Reply:
> Just curious - What is the value of adding the Experimental prefix here?

I'd have gone with "Liudmila told me to."   😄

Reply:
we also have a bunch of unstable conventions for Azure client libs (cosmos, messaging, GenAI), so yeah, we need to communicate to users that the dashboards, alerts and queries they wrote on top of our telemetry can be broken at any moment (even though we try not to).

We're following the pattern .NET is doing here https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-builtin-activities#http-client-request-wait-for-connection-experimental